StandaloneMmPkg/Library/StandaloneMmMemLib/X86StandaloneMmMemLibInternal.c

```diff
@@ -74,66 +76,192 @@ MmMemLibInternalCalculateMaximumSupportAddress (
 }
 
 /**
-  Initialize cached Mmram Ranges from HOB.
+  Merge the overlapped or continuous ranges in input MemoryRange. This function is to optimize
+  the process of checking whether a buffer range belongs to the range reported by resource HOB,
+  since the buffer to be checked may be covered by multi resource HOB.
+
+  @param[in, out]  MemoryRange            A pointer to the NonMmramRanges reported by resource HOB.
+  @param[in, out]  MemoryRangeSize        A pointer to the size, in bytes, of the MemoryRange buffer.
+                                          On input, it is the size of the current memory map.
+                                          On output, it is the size of new memory map after merge.
+**/
+STATIC
+VOID
+MergeOverlappedOrContinuousRanges (

```

Comment:
we can create another PR to update the MergeMemoryMapForNotPresentEntry() in SMM CPU driver to align with this change:

1. handler the overlap case. i guess uefimem also might has the overlap case.
2. refine the copy flow.

Reply:
Yes I can create the PR later